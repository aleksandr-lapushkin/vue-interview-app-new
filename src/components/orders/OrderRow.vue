<template>
  <tr>
    <td class="order-id">{{ order.id }}</td>
    <td class="order-title">{{ order.title }}</td>
    <td class="order-status">{{ order.status }}</td>
  </tr>
</template>

<script>
export default {
  name: "OrderRow",
  props: ["order"]
};
</script>

<style scoped>
td {
  padding: 1em;
}
tr:hover {
  background: #eee;
  transition-duration: 300ms;
  cursor: pointer;
}
</style>
