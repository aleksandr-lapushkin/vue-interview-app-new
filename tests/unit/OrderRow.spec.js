import { shallowMount } from "@vue/test-utils";
import OrderRow from "../../src/components/orders/OrderRow.vue";

test("OrderRow has correct values", () => {
  const wrapper = shallowMount(OrderRow, {
    propsData: {
      order: { id: 123, title: "hello world", status: "PROCESSING" }
    }
  });
  expect(wrapper.find(".order-id").text()).toEqual("123");
  expect(wrapper.find(".order-title").text()).toEqual("hello world");
  expect(wrapper.find(".order-status").text()).toEqual("PROCESSING");
  expect(wrapper.html()).toMatchSnapshot();
});
