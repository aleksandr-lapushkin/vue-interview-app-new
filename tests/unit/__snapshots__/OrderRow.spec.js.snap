// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderRow has correct values 1`] = `
<tr>
  <td class="order-id">123</td>
  <td class="order-title">hello world</td>
  <td class="order-status">PROCESSING</td>
</tr>
`;
